### PR TITLE
Remove Legacy Builtin Price Estimator

### PR DESCRIPTION
**Pull request by nlordell**
This is a follow up to #1715 where we added a new `--price-estimation-legacy-solvers` flag to specify legacy HTTP solvers to use as price estimators. As such, the "builtin" estimators `Quasimodo`, `Yearn` and `Raven` are no longer needed.

This PR deprecates the aforementioned builtin estimators in favor of the new `--price-estimation-legacy-solvers` flag.

### Test Plan

Rust CI - we are removing code, so nothing to test!

### Release notes

https://github.com/cowprotocol/infrastructure/pull/596 already does the required changes in order to no longer rely on the "builtin" versions of these legacy price estimators - those changes have already been applied.

### CODE REVIEW

**Review by MartinquaXD — APPROVED** [review]
Nice 🧹

**Review by sunce86 — APPROVED** [review]

**Comment by nlordell** [comment]
Before merging this PR, I would need some approvals in https://github.com/cowprotocol/infrastructure/pull/596 as well.

**Pull request merged**
